nameserver/server_test.go

```diff
@@ -56,6 +63,16 @@ func TestDNSServer(t *testing.T) {
 	wt.AssertType(t, r.Answer[0], (*dns.A)(nil), "DNS record")
 	wt.AssertEqualString(t, r.Answer[0].(*dns.A).A.String(), testAddr1, "IP address")
 
+	// Retry the query with the TCP client
+	tr, _, err := tc.Exchange(m, dnsAddr)
+	wt.AssertNoErr(t, err)
+	wt.AssertStatus(t, tr.Rcode, dns.RcodeSuccess, "DNS response code (TCP)")
+	wt.AssertEqualInt(t, len(tr.Answer), 1, "Number of answers (TCP)")
+	wt.AssertType(t, tr.Answer[0], (*dns.A)(nil), "DNS record (TCP)")
+	wt.AssertEqualString(t, tr.Answer[0].(*dns.A).A.String(), testAddr1, "IP address (TCP)")
+	// TODO: look for some way of testing the TCP fallback for truncated responses: it seems there is
+	// TODO: no client lib in Go that can do that...
```

Comment:
I think you should test a large response, say 8K, then you can try it over UDP and expect truncation, then try it over TCP.